mpd: move to pkgs/by-name

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SuperSandro2000 commented:
It would be easier to review if the deprecation of mpdWithFeatures would be done in an extra commit before the move because now it is just one big commit which changed nearly everything. 😅

doronbehar commented:
> It would be easier to review if the deprecation of mpdWithFeatures would be done in an extra commit before the move because now it is just one big commit which changed nearly everything. 😅

I'm too in favor of decoupling changes that can logically be decomposed, but in this case that's almost impossible - I wanted to get rid of the lines in `all-packages.nix` all at once and that forced me to do it all together.

ambroisie commented:
> I'm too in favor of decoupling changes that can logically be decomposed, but in this case that's almost impossible - I wanted to get rid of the lines in all-packages.nix all at once and that forced me to do it all together.

I would say you should first get rid of `mpdWithFeatures`, then the remaining lines.

As it is, I agree with Sandro, this is not reviewable without a great deal of effort. I'd rather be able to first review the move to `mpd.overrides`, then the (presumably much simpler, likely just a few file renames) move to `pkgs/by-name`.

doronbehar committed:
mpd: no with lib; in meta

doronbehar committed:
mpd: use finalAttrs

doronbehar committed:
mpd-small: move to pkgs/by-name

doronbehar committed:
mpd: promote features list to top level arguments

From some reason, using mpdWithFeature = mpd.override inside
mpd/default.nix causes an evaluation error

doronbehar committed:
mpd: define simpler, using callPackage (singular)

doronbehar committed:
mpd: format expression

doronbehar committed:
mpd: move to pkgs/by-name

doronbehar commented:
Well, if I'm allowing myself to commit changes that aren't formatted, (and if you insist), it is possible to decouple a the changes a little bit.

ambroisie reviewed (approved):
Thank you @doronbehar for splitting into smaller commits. LGTM.

doronbehar commented:
Will give @SuperSandro2000 and others a few more days to reply.

SuperSandro2000 reviewed (approved):
LGTM
